### PR TITLE
test, cloudinit: Use "json" annotation instead of "yaml"

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
The networkData from cloud init is being rendered wrongly [1] since the actual yaml
parser that we use sigs.k8s.io/yaml don't understand the "yaml" annotation [2], using
"json" annotation fix the issue.

It also refactor some tests that were configuring ipv6 twice first with cloud-init then at expecter.

[1]  
```yaml
cloudInitNoCloud:
        networkData: |
          Ethernets:
            eth0:
              AcceptRA: null
              Addresses:
              - fd10:0:2::2/120
              DHCP4: true
              DHCP6: null
              DHCPIdentifier: ""
              Gateway4: ""
              Gateway6: fd10:0:2::1
              MACAddress: ""
              MTU: 0
              Nameservers:
                Addresses:
                - 10.96.0.10
                Search:
                - default.svc.cluster.local
                - svc.cluster.local
                - cluster.local
              Routes: null
          Version: 2
```

[2] https://play.golang.org/p/S0Pok-gGeQH

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by EdDev — APPROVED** [review]
Thank you and sorry you had to waste time on this.

**Comment by qinqon** [comment]
I think we have to discover why tests are passing without it.

**Comment by qinqon** [comment]
Added bazel BUILD files

File: Makefile [hunk 1]
```diff
@@ -154,6 +154,9 @@ olm-push:
 bump-kubevirtci:
 	./hack/bump-kubevirtci.sh
 
+foo:
+	go test ./tests/...
+
```
Comment (EdDev) [inline Makefile:159]: Changes in this file are probably leftovers.

File: tests/errors/panic.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package errors
+
+func PanicOnError(err error) {
```
Comment (EdDev) [inline tests/errors/panic.go:3]: If you created this one, why didn't you used it in the console/login.go ?
But I would prefer to see this in a different commit or better, in a different PR.

File: tests/libvmi/pod.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	"kubevirt.io/kubevirt/pkg/controller"
+)
+
+func PanicOnError(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+func GetPodByVirtualMachineInstance(vmi *v1.VirtualMachineInstance, namespace string) *k8sv1.Pod {
```
Comment (EdDev) [inline tests/libvmi/pod.go:20]: How is this related to libnet ?

File: tests/login.go [hunk 4]
```diff
@@ -27,174 +25,98 @@ func LoginToCirros(vmi *v1.VirtualMachineInstance) error {
 // Alpine test VM
 func LoggedInCirrosExpecter(vmi *v1.VirtualMachineInstance) (expect.Expecter, error) {
 	virtClient, err := kubecli.GetKubevirtClient()
```
Comment (EdDev) [inline tests/login.go:27]: Leftover?

**Review by EdDev — CHANGES_REQUESTED** [review]
This change can be split into two main changes:
- Fix the yaml annotations (1st commit).
- Use the cloud-init network config instead of the loggin IPv6 configuration.

If I understand correctly, the 2nd one is to confirm the 1st works.
It looks too big change to fit under the same intent here and I would have preferred to see it in a different PR. I hope we can combine it with [this one](https://github.com/kubevirt/kubevirt/pull/4362).

Regarding confirming that the 1st really works, I think we have a production issue.
If we take this simple [test](https://github.com/kubevirt/kubevirt/blob/88ddf9afc21dbbc8fe0260ecbf7d5514e35cdee6/tests/network/primary_pod_network.go#L105), it does no login at all, and still it checks that the pod and VMI reported IP/s match. But as we just found out when fixing the annotation, there is no IPv6 on the VM itself, so even though we have GA deployed, we are not reading from it the IP/s correctly, otherwise I would have expected the test to fail.

**Comment by oshoval** [comment]
didnt this PR was pure json / yaml change ? 
and the other was the one with all the ipv6 changes?

File: tests/console/login.go [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+	"kubevirt.io/kubevirt/pkg/util/net/dns"
+)
+
+func PanicOnError(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+// LoginToCirros call LoggedInFedoraExpecter but does not return the expecter
```
Comment (fgimenez) [inline tests/console/login.go:24]: `LoginToCirros` is still present in `tests/login.go`, these changes are duplicating it, shouldn't it be removed from there? Or maybe update the functions in `tests/console/login.go` to use the version in `tests/login.go`. Also, the version in this PR is not up to date, see https://github.com/kubevirt/kubevirt/blob/master/tests/login.go#L19-L26.

**Pull request closed without merging**